src/views/Dashboard/Customer/Home/MyTest/index.vue

```diff
@@ -31,30 +75,147 @@
 <script>
 import { layersIcon, noteIllustration } from "@/common/icons"
 import StakingServiceTab from "./StakingServiceTab.vue"
+import DataTable from "@/common/components/DataTable"
+import Button from "@/common/components/Button"
+import { mapState } from "vuex"
+import { ordersByCustomer, getOrdersData } from "@/common/lib/polkadot-provider/query/orders"
+import { queryLabsById } from "@/common/lib/polkadot-provider/query/labs"
+import { queryServicesById } from "@/common/lib/polkadot-provider/query/services"
+import localStorage from "@/common/lib/local-storage"
+import dataTesting from "./dataTesting.json"
 
 export default {
   name: "MyTest",
 
-  components: { StakingServiceTab },
+  components: { 
+    StakingServiceTab,
+    DataTable,
+    Button
+  },
 
   data: () => ({ 
     layersIcon, 
     noteIllustration, 
     cardBlock: false,
     documents: null,
-    tabs: null
+    tabs: null,
+    isReady: false,
+    isRegistered: false,
+    isProcessed: false,
+    orderHistory: [],
+    headers: [
+      { text: "Service Name", value: "service_info.name", sortable: true },
+      { text: "Lab Name", value: "lab_info.name", sortable: true },
+      { text: "Order Date", value: "created_at", sortable: true },
+      { text: "Last Update", value: "updated_at", sortable: true },
+      { text: "Test Status", value: "status", sortable: true },
+      {
+        text: "Actions",
+        value: "actions",
+        sortable: false,
+        align: "center",
+        width: "5%"
+      }
+    ]
   }),
 
   mounted() {
-    window.addEventListener("resize", () => {
-      if (window.innerWidth <= 959) this.cardBlock = true
-      else this.cardBlock = false
-    })
+    this.onSearchInput();
   },
 
+  async created() {},
+
   methods: {
     toRequestTest() {
       this.$router.push({ name: "customer-request-test-select-lab"})
+    },
+
+    async onSearchInput() {
+      this.orderHistory = dataTesting.data.map(result => ({
+        ...result._source,
+        id: result._id,
+        updated_at: new Date(parseInt(result._source.updated_at)).toLocaleDateString(),
+        created_at: new Date(parseInt(result._source.created_at)).toLocaleDateString(),
+        timestamp: parseInt(result._source.created_at)
+      }))
+    },
+
+    setStatusColor(status) {
+      let colors = Object.freeze({
+        REGISTERED: "#44921A",
+        PAID: "#E27625",
+        RESULTREADY: "#5640A5",
+        REJECTED: "#9B1B37",
+        SUBMITEDASDATABOUNTY: "#5640A5"
+      })
+      return { color: colors[status.toUpperCase()] }
+    },
+
+    async getOrderHistory() {
+      this.isLoadingOrderHistory = true;
+      try {
+        this.orderHistory = [];
+        const address = this.wallet.address;
+        const listOrderId = await ordersByCustomer(this.api, address);
+        var lengthMax = 3;
+        if (listOrderId != null) {
```

Comment:
Too many conditionals inside one another, please change how you arrange the conditionals or separate into smaller functions to be more maintainable.